library/arith.lisp

```diff
@@ -677,6 +698,86 @@ The fields are defined as follows:
 
 (%define-standard-complex-instances Fraction)
 
+;;;; Floating
+
+(coalton-toplevel
+  (define-class (Exponent :a)
+    "Exponential functions where exp is exponentiation of Euler's number and
+(exp (+ a b)) = (* (exp a) (exp b)) and (exp 0) = 1."
+    (exp (:a -> :a)))
+
+  (define-class (Exponent :a => Logarithm :a)
```

Comment:
No specific examples, but _if_ we end up splitting the class like this I think exp would be the most useful function. I guess for the cases were either ln(X) doesn't converge but e^X does - it could just throw an error (or in the case were the implementer is too lazy).